### PR TITLE
ThemeOwner: Extract get_owner_node_theme

### PR DESCRIPTION
**Pull request by quentinguidee**
Fix code duplication (x5) in `theme_owner.cpp`

### CODE REVIEW

File: scene/theme/theme_owner.cpp [hunk 1]
```diff
@@ -408,3 +363,19 @@ int ThemeOwner::get_theme_default_font_size() {
 	}
 	return ThemeDB::get_singleton()->get_fallback_font_size();
 }
+
+Ref<Theme> ThemeOwner::_get_owner_node_theme(Node *p_owner_node) const {
+	Ref<Theme> owner_theme;
+
+	Control *owner_c = Object::cast_to<Control>(p_owner_node);
+	if (owner_c) {
+		owner_theme = owner_c->get_theme();
```
Comment (YuriSizov) [inline scene/theme/theme_owner.cpp:372]: I think we can simplify this and return immediately in both cases. It was more awkward to write it like that when it was a part of a bigger method, but in a dedicated method we can remove unnecessary casts if we get a hit.

Then the last statement would just be `return Ref<Theme>();`.

**Review by YuriSizov — APPROVED** [review]
Yeah, I wanted to do this, but I didn't feel like it was worth making a PR over it. But that is sure appreciated 🙃

**Comment by quentinguidee** [comment]
Thanks for the review! Should be ok now.

> Yeah, I wanted to do this, but I didn't feel like it was worth making a PR over it. But that is sure appreciated 🙃

For me there is a big issue of duplication across the entire codebase, and I think all cases should be addressed, starting with the small ones.

Every time I jump into the refactor of a class, it turns out that the code I'm refactoring is also duplicated in another place, and sometimes it's even several hundred lines. If someone refactors one chunk and not the other by mistake, the two chunks become unlinked and will probably never be refactored.

There is an issue about duplicated code here #47397 with no likes and I'm surprised it didn't even start a conversation. Is there another conversation about this elsewhere?

**Comment by quentinguidee** [comment]
Oh wait should I switch both if statements now since I early return?

**Comment by YuriSizov** [comment]
> Oh wait should I switch both if statements now since I early return? First the window then the control or does it not matter?

It doesn't matter, no node can be a Window and a Control at the same time, so this wouldn't affect anything. Besides, it's more likely to be a Control.

> There is an issue about duplicated code here #47397 with no likes and I'm surprised it didn't even start a conversation. Is there another conversation about this elsewhere?

Code duplication is not inherently bad, so there is no focused effort to reduce it. But it's okay to PR patches in specific areas where it makes sense, like here.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**